Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -0,0 +1,242 @@
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Data.SqlClient;
+
+namespace Gordon360.Services.RecIM
+{
+    public class ParticipantService : IParticipantService
+    {
+        private readonly CCTContext _context;
+        private readonly IAccountService _accountService;
+
+        public ParticipantService(CCTContext context, IAccountService accountService)
+        {
+            _accountService = accountService;
+            _context = context;
+        }
+        public IEnumerable<LookupViewModel> GetParticipantLookup(string type)
+        {
+            if (type == "status")
+            {
+                var res = _context.ParticipantStatus
+                            .Select(s => new LookupViewModel
+                            {
+                                ID = s.ID,
+                                Description = s.Description
+                            })
+                            .AsEnumerable();
+                return res;
+            }
+            if (type == "activitypriv")
+            {
+                var res = _context.PrivType
+                            .Select(s => new LookupViewModel
+                            {
+                                ID = s.ID,
+                                Description = s.Description
+                            })
+                            .AsEnumerable();
+                return res;
+            }
+            return null;
+        }
+        public ACCOUNT GetAccountByParticipantID(int ID)
+        {
+            return _context.ACCOUNT
+                        .FirstOrDefault(a => a.gordon_id == $"{ID}");
+        }
+        public ParticipantExtendedViewModel GetParticipantByUsername(string username)
+        {
+            var account = _accountService.GetAccountByUsername(username);
+            var participant = _context.Participant
+                                .Where(p => p.Username == username)
+                                .Select(p => new ParticipantExtendedViewModel
+                                {
+                                    Username = username,
+                                    Email = account.Email,
+                                    Status = _context.ParticipantStatusHistory
+                                                .Where(psh => psh.ParticipantUsername == username)
+                                                .OrderByDescending(psh => psh.ID)
+                                                .Take(1)
+                                                    .Join(_context.ParticipantStatus,
+                                                        psh => psh.StatusID,
+                                                        ps => ps.ID,
+                                                        (psh, ps) => ps.Description)
+                                                .FirstOrDefault(),
```

Comment:
Can this be condensed to the following?:
```suggestion
                                    Status = p.ParticipantStatusHistory
                                    .OrderByDescending(psh => psh.ID)
                                    .Take(1).Status.Description,
```

Reply:
Yep, this was written when I was unaware of how much LINQ actually did for me.